### DISCUSSION TITLE
Use MemberData and a String at the parameter level

### DISCUSSION THREAD

**Discussion by DirttDiver** [xunit/xunit#2407] [Question] [answered]
Does anyone know if it's possible to use another datatype along with MemberData? I have parameterized tests which are working, but I have since needed to add a test initialser method within each test (for various reasons)

I need to be able to use MemberData the same way as inline data

Currently this is my setup and it works perfectly. I have a list of objects returned and each object can run as an individual test.

        [Trait("Category", "FE")]
        [Trait("Category", "DocumentManagement")]
        [Trait("Category", "CriticalDocuments")]
        [Theory]
        [MemberData(nameof(FilterTypes))]
        [MemberData(nameof(FilterTypes))]
        public void Test_Filter(int id, TheoryWrapper<PageFilters> filterParams)

        {
            critical_documents.AssertFilterIsNotEmtpy(filterParams.Object.filter_locater);
        }
I do however need to be able to add a string at the test parameter level to allow for some other method to work correctly.

e.g.

        [Trait("Category", "FE")]
        [Trait("Category", "DocumentManagement")]
        [Trait("Category", "CriticalDocuments")]
        [Theory]
        [MemberData(nameof(FilterTypes), "user1")]
        [MemberData(nameof(FilterTypes), "user2")]
        public void Test_Filter(int id, TheoryWrapper<PageFilters> filterParams, string userType)

        {
            critical_documents.InitializeTest(user_type: userType);
            critical_documents.AssertFilterIsNotEmtpy(filterParams.Object.filter_locater);
        }
any ideas on the syntax or if this is possible? Thanks

**Comment by bradwilson** [comment, marked as answer]
This behavior is actually already available. You need to convert `FilterTypes` to a method, and then the parameters that you're passing (`"user1"`, `"user2"`) will be passed as arguments to the method. Example:

```csharp
public static IEnumerable<object[]> FilterTypes(string userName)
{
    // ... generate data based on the userName parameter value ...
}
```
